scheduler/executor/interfaces/schedule_checkpointer.go

```diff
@@ -0,0 +1,9 @@
+package interfaces
+
+import "context"
+
+type ScheduleCheckPointer interface {
```

Comment:
```suggestion
type CheckPointer interface {
```
rename the folder to schedule or core